### PR TITLE
Conode add update

### PR DESCRIPTION
**Pull request by ineiti**
Update and putting the port to 7770

### CODE REVIEW

**Comment by nikkolasg** [comment]
OK to me.

File: app/conode/exit_conodes.sh [hunk 1]
```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 
-HOSTS=$( cat config.toml | grep Hosts | sed -e "s/.*\[\"\(.*\)\"\]/\1/" | perl -pe "s/\", \"/\n/g" )
+CONFIG=config.toml
+if [ "$1" ]; then
+  CONFIG=$1/$CONFIG
+fi
+HOSTS=$( grep Hosts $CONFIG | sed -e "s/.*\[\"\(.*\)\"\]/\1/" | perl -pe "s/\", \"/\n/g" | tail -r )
```
Comment (Liamsi) [inline app/conode/exit_conodes.sh]: Note: after merging we've recognized that `tail -r` won't work on most linux machines...

File: app/conode/update.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        ./update.sh update_version
+        exec ./start-conode.sh run
+        ;;
+    esac
+}
+
+# Fetches the latest version and untars it here
+update(){
+  if [ ! -e NO_UPDATE ]; then
+    RELEASE=$( wget -q -O- https://github.com/dedis/cothority/releases/latest | grep DeDiS/cothority/releases/download | sed -e "s/.*href=.\(.*\). rel.*/\1/" )
```
Comment (Liamsi) [inline app/conode/update.sh]: `grep -i DeDiS/cothority/releases/download`

File: app/conode/start-conode.sh [hunk 3]
```diff
@@ -33,45 +48,27 @@ main(){
...
     echo $0 run
+    echo "or if you want to manually update:"
+    echo $0 update
     echo
     ;;
   esac
 }
 
-update(){
-  RELEASE=$( wget -q -O- https://github.com/dedis/cothority/releases/latest | grep DeDiS/cothority/releases/download | sed -e "s/.*href=.\(.*\). rel.*/\1/" )
```
Comment (Liamsi) [inline app/conode/start-conode.sh]: case insenstive grep, too

File: app/conode/exit_conodes.sh (hunk 1 above)
Comment (ineiti) [inline app/conode/exit_conodes.sh]: Le 16/12/2015 14:21, Ismail a écrit :
>
> In app/conode/exit_conodes.sh 
> <https://github.com/dedis/cothority/pull/142#discussion_r47774281>:
>
> > @@ -1,6 +1,10 @@
> >  #!/bin/bash
> >
> > -HOSTS=$( cat config.toml | grep Hosts | sed -e "s/.*\[\"\(.*\)\"\]/\1/" | perl -pe "s/\", \"/\n/g" )
> > +CONFIG=config.toml
> > +if [ "$1" ]; then
> > +  CONFIG=$1/$CONFIG
> > +fi
> > +HOSTS=$( grep Hosts $CONFIG | sed -e "s/.*\[\"\(.*\)\"\]/\1/" | perl -pe "s/\", \"/\n/g" | tail -r )
>
> Note: after merging we've recognized that |tail -r| won't work on most 
> linux machines...
>
>
There is `tac` - but I think that doesn't exist everywhere, neither. At 
least it doesn't on a mac. I wanted to exit the leaf-nodes as first 
nodes, then the root-node. But in fact it doesn't do it hierarchly 
anyway. So you might as well just throw away the `tail -r`.

Comment (Liamsi) [inline app/conode/exit_conodes.sh]: Thanks @ineiti :-)

**Pull request merged**